src/playground/system_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns playground.system-test
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            [io.pedestal.test :refer [response-for]]
+            [com.stuartsierra.component :as component]
+            [clojure.test :refer :all]
+            [user]
+            [playground.server]
+            [playground.service :as service]))
+
+
+(def url-for (route/url-for-routes
+              (route/expand-routes service/routes)))
+
+(defn service-fn
+  [system]
+  (get-in system [:pedestal :service ::http/service-fn]))
+
+(defmacro with-system
+  [[bound-var binding-expr] & body]
+  `(let [~bound-var (component/start ~binding-expr)]
+     (try
+       ~@body
+       (finally
+         (component/stop ~bound-var)))))
+
+(deftest greeting-test
+  (with-system [sut (user/dev-system)]                       ;; <1>
+    (let [service               (service-fn sut)                 ;; <2>
```

Comment:
Remove `with-system [sut (user/dev-system)]` entirely. You should never invoke (user/dev-system) by hand; that is Component's job.

Do this instead:

```
(let [system com.stuartsierra.component.repl/system ;; com.stuartsierra.component.repl/system holds your system after a `(reset)`
      service (service-fn system)]
  ...)
```

Reply:
thanks!